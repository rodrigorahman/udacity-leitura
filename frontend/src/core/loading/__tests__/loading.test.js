import React from "react";
import ReactDOM from "react-dom";
import LoadingComponent from "../loading.component";

import { shallow } from "enzyme";
import "core-js/es6/map";
import "core-js/es6/set";

it("renders without crashing", () => {
  let show = true;
  expect(shallow(<LoadingComponent show={show} />)).toMatchSnapshot();
});

it("check open loading", () => {
  let show = false;
  let loading = shallow(<LoadingComponent show={show} />);
  expect(loading.find("#myModal")).toHaveLength(0);
});


it("check open loading", () => {
  let show = true;
  let loading = shallow(<LoadingComponent show={show} />);
  expect(loading.find("#myModal")).toHaveLength(1);
});
