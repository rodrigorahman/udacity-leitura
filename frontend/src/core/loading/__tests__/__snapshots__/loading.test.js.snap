// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="modalLoading"
  id="myModal"
>
  <div
    className="lds-css ng-scope centralizado"
    style={
      Object {
        "height": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="lds-eclipse"
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div />
    </div>
  </div>
  <div
    className="centralizado text"
  >
    Carregando!
  </div>
</div>
`;
